File: client/src/main/java/com/jraska/github/client/ui/ShortcutHandlerActivity.java
package com.jraska.github.client.ui;

import android.os.Bundle;

import okhttp3.HttpUrl;

public class ShortcutHandlerActivity extends BaseActivity {

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);

    ShortcutHandlerModel handlerModel = viewModel(ShortcutHandlerModel.class);

    HttpUrl httpUrl = HttpUrl.parse(getIntent().getData().toString());
    handlerModel.handleDeepLink(httpUrl);

    finish();
  }
}

File: client/src/main/java/com/jraska/github/client/ui/ShortcutHandlerActivity.kt
package com.jraska.github.client.ui

import android.os.Bundle
import com.jraska.github.client.inputUrl
import com.jraska.github.client.core.android.viewModel

class ShortcutHandlerActivity : BaseActivity() {

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)

    val handlerModel = viewModel(ShortcutHandlerModel::class.java)

    val httpUrl = inputUrl()
    handlerModel.handleDeepLink(httpUrl)

    finish()
  }
}
